src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,214 @@
+};
+
+const childChecked = (treeNode: TreeNodeItem): boolean => {
+    return treeNode.children ? treeNode.children.some(child => childChecked(child)) : isChecked(treeNode);
+};
+
+const allChildrenChecked = (treeNode: TreeNodeItem): boolean  => {
+    return treeNode.children ? treeNode.children.every(child => allChildrenChecked(child)) : isChecked(treeNode);
+};
+
+const initTreeNodeById = (groups: readonly Schemas.Facet[], filters: EventLogCustomFilter[]) => {
+    const init: TreeNodeDict = {};
+    groups.forEach(group => {
+        const currentFilter = filters.find(filter => filter.bundleId === group.name);
+        const currentFilterChipValues = currentFilter?.chips?.map(chip => chip.value);
+
+        const items = group.children as Schemas.Facet[];
+        const checkAll = items.length !== 0 ? items.every(item => !!currentFilterChipValues?.includes(item.name)) : false;
+        init[group.name] = {
+            id: group.name,
+            name: group.displayName,
+            checkProps: { checked: checkAll || (!currentFilter ? false : null) },
+            children: items.length !== 0 ? items.map(item => ({
+                id: item.name,
+                name: item.displayName,
+                checkProps: { checked: checkAll || currentFilterChipValues?.includes(item.name) }
+            })) : undefined
+        };
+    });
+
+    return init;
+};
+
+export const EventLogTreeFilter: React.FunctionComponent<EventLogTreeFilterProps> = (props) => {
+    const { groups, placeholder, filters, updateFilters } = props;
+
+    const initialize = React.useMemo(() => initTreeNodeById(groups, filters), [ groups, filters ]);
+
+    const [ treeNodeById, setTreeNodeById ] = React.useState<TreeNodeDict>(initialize);
+    const [ isToggled, setIsToggled ] = React.useState(false);
+
+    const treeDataArray = React.useMemo(() => !!treeNodeById ? Object.values(treeNodeById) : [], [ treeNodeById ]);
+
+    React.useEffect(() => {
+        if (groups.length !== 0) {
+            setTreeNodeById(produce((prev) => {
+                // In case Bundles request is pending during Tree View render
+                if (Object.keys(prev).length === 0) {
+                    return initialize;
+                }
+                else if (filters.length === 0) {
+                    return initialize;
+                }
+                else {
+                    filters.forEach(activeFilter => {
+                        const treeNode = prev[activeFilter.bundleId];
+                        const activeChips = activeFilter.chips.map(chip => chip.value);
+
+                        treeNode.children?.forEach(childNode => {
+                            childNode.checkProps.checked = activeChips.includes(childNode.id);
+                        });
+
+                        if (allChildrenChecked(treeNode)) {
+                            treeNode.checkProps.checked = true;
+                        }
+                        else if (childChecked(treeNode)) {
```

Comment:
I am definitely in the minority when it comes to grouping my conditional blocks... I think having something consistent across the codebase is the most important part.. unfortunately, i have too much muscle memory to do it that way to begin with. Will change to suggested structure and second the idea of having the linter enforce it, because I know I'll forget again.